front/pages/w/[wId]/builder/data-sources/public-urls.tsx

```diff
@@ -0,0 +1,207 @@
+    (ds) => ds.connectorProvider === "webcrawler"
+  );
+
+  return {
+    props: {
+      user,
+      owner,
+      subscription,
+      plan,
+      readOnly,
+      dataSources,
+      gaTrackingId: GA_TRACKING_ID,
+    },
+  };
+};
+
+export default function DataSourcesView({
+  user,
+  owner,
+  subscription,
+  plan,
+  readOnly,
+  dataSources,
+  gaTrackingId,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+  const [showDatasourceLimitPopup, setShowDatasourceLimitPopup] =
+    useState(false);
+
+  const {
+    submit: handleCreateDataSource,
+    isSubmitting: isSubmittingCreateDataSource,
+  } = useSubmitFunction(async () => {
+    // Enforce plan limits: DataSources count.
+    if (
+      plan.limits.dataSources.count != -1 &&
+      dataSources.length >= plan.limits.dataSources.count
+    ) {
+      setShowDatasourceLimitPopup(true);
+    } else {
+      void router.push(`/w/${owner.sId}/builder/data-sources/new-public-url`);
+    }
+  });
+
+  return (
+    <AppLayout
+      subscription={subscription}
+      user={user}
+      owner={owner}
+      gaTrackingId={gaTrackingId}
+      topNavigationCurrent="admin"
+      subNavigation={subNavigationAdmin({
+        owner,
+        current: "data_sources_url",
+      })}
+    >
+      <Page.Vertical gap="xl" align="stretch">
+        <Page.Header
+          title="Public URLs"
+          icon={LinkIcon}
+          description="Manage public URLs as data sources for the workspace."
+        />
+
+        {dataSources.length > 0 ? (
+          <div className="relative">
+            <Page.SectionHeader
+              title=""
+              description=""
+              action={
+                !readOnly
+                  ? {
+                      label: "Add a public URL",
+                      variant: "primary",
+                      icon: PlusIcon,
+                      onClick: async () => {
+                        await handleCreateDataSource();
+                      },
+                      disabled: isSubmittingCreateDataSource,
+                    }
+                  : undefined
+              }
+            />
+            <Popup
+              show={showDatasourceLimitPopup}
+              chipLabel={`${plan.name} plan`}
+              description={`You have reached the limit of data sources (${plan.limits.dataSources.count} data sources). Upgrade your plan for unlimited datasources.`}
+              buttonLabel="Check Dust plans"
+              buttonClick={() => {
+                void router.push(`/w/${owner.sId}/subscription`);
+              }}
+              onClose={() => {
+                setShowDatasourceLimitPopup(false);
+              }}
+              className="absolute bottom-8 right-0"
+            />
+          </div>
+        ) : (
+          <EmptyCallToAction
+            label="Create a new Folder"
+            onClick={async () => {
+              await handleCreateDataSource();
```

Comment:
same here, you don't need all this wrapping right ? you can just pass `handleCreateDataSource`.